src/elasticsearch/codec.go

```diff
@@ -48,9 +48,14 @@ func (c basicCodec) EncodeElasticRecords(records []*models.Record) ([]*models.El
 }
 
 func (c basicCodec) getDatabaseIndex(record *models.Record) (string, error) {
-	indexPrefix := c.config.Index
-	if indexPrefix == "" {
-		indexPrefix = record.Topic
+	indexName := c.config.Index
+	if indexName == "" {
+		indexName = record.Topic
+	}
+
+	indexPrefix := c.config.IndexPrefix
+	if indexPrefix != "" {
+		indexName = indexPrefix + indexName
 	}
```

Comment:
This `if` clause is unnecessary.
Simply `indexName = c.config.IndexPrefix + indexName` would work

Reply:
Cool! I will make the requested changes. Thanks!